tox.ini

```diff
@@ -0,0 +1,92 @@
+[tox]
+envlist = py{39,310,311,312,313,314,py310}
+labels =
+    check = typing, gen_exports, type_completeness, pip_compile
+    cython = py39-cython2,py39-cython,py311-cython2,py313-cython
+
+# TODO:
+# * environment to check coverage
+# * replace ci.sh
+#   * --verbose --durations=10
+#   * -p _trio_check_attrs_aliases
+#   * mypy cache
+#   * LSP
+#   * apport
+# * use tox in CI
+# * switch to nox?
+# * move to pyproject.toml?
+#   * this means conditional deps need to be replaced
+
+# protip: install uv-tox for faster venv generation
+
+[testenv]
+# use wheels instead of sdist, significantly faster install
+package = wheel
+wheel_build_env = .pkg
+deps =
+    hypothesis: hypothesis
+    -r test-requirements.txt
+set_env =
+    slow: TOX_RUN_SLOW = '--run-slow'
+commands =
+    pytest {env:TOX_RUN_SLOW:} {posargs}
+
+[testenv:no_test_requirements]
+commands =
+    pytest --skip-optional-imports {posargs}
+
+[testenv:docs]
+deps =
+    -r docs-requirements.txt
+# base_python synced with .readthedocs.yml
```

Comment:
We can also make RTD just call tox in a follow-up. This is what I do these days.